src/test/java/com/soolsul/soolsulserver/common/annotation/MockCustomUser.java

```diff
@@ -0,0 +1,28 @@
+package com.soolsul.soolsulserver.common.annotation;
+
+import com.soolsul.soolsulserver.common.factory.MockCustomUserSecurityContextFactory;
+import org.springframework.core.annotation.AliasFor;
+import org.springframework.security.test.context.support.TestExecutionEvent;
+import org.springframework.security.test.context.support.WithSecurityContext;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@WithSecurityContext(factory = MockCustomUserSecurityContextFactory.class)
+public @interface MockCustomUser {
+
+    String value() default "user";
+
+    String username() default "";
+
+    String[] roles() default {"USER"};
+
+    String password() default "password";
+
+    @AliasFor(annotation = WithSecurityContext.class)
+    TestExecutionEvent setupBefore() default TestExecutionEvent.TEST_METHOD;
```

Comment:
아 이거는 기존 WithSecurityContext의 setupBefore()를 사용중임을 명시적으로 오버라이딩 하고 있습니다.
테스트 전에 삽입되어야 함을 강조하는 부분이죠!

다만 없어도 이미 WithSecurityContext 때문에 테스트 전에 삽입됩니다.
```
public @interface WithSecurityContext {

	Class<? extends WithSecurityContextFactory<? extends Annotation>> factory();

	/**
	 * Determines when the {@link SecurityContext} is setup. The default is before
	 * {@link TestExecutionEvent#TEST_METHOD} which occurs during
	 * {@link org.springframework.test.context.TestExecutionListener#beforeTestMethod(TestContext)}
	 * @return the {@link TestExecutionEvent} to initialize before
	 * @since 5.1
	 */
	TestExecutionEvent setupBefore() default TestExecutionEvent.TEST_METHOD;
}
```

Reply:
테스트 전에 삽입된다는 것을 @WithSecurityContext 어노테이션을 통해서 파악할 수 있어서 이 부분은 제거하는 방식이 좋지 않을까 생각합니다

Reply:
넵 제거하겠습니다!